### PR TITLE
API extension to retrieve provider parameters

### PR DESCRIPTION
**Pull request by sciascid**

### CODE REVIEW

File: wsrep_config_service.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                                         void* context);
+
+/**
+ * Get configuration parameters exposed by the provider.
+ *
+ * @param wsrep    pointer to provider handle
+ * @param cb       function pointer for callback
+ * @param context  application context passed to callback
+ */
+typedef void (*wsrep_get_parameters_fn) (wsrep_t* wsrep,
```
Comment (temeo) [inline wsrep_config_service.h:89]: This function returns void. How errors are propagated?

Comment (temeo) [reply wsrep_config_service.h:89]: We probably need to handle both error in provider and error on server side during callback call.

File: wsrep_config_service.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    WSREP_PARAM_TYPE_DOUBLE     \
+    )
+
+typedef struct wsrep_parameter
+{
+  int flags;
+  const char* name;
+  union {
+    bool as_bool;
+    int64_t as_integer;
```
Comment (temeo) [inline wsrep_config_service.h:67]: Should we have unsigned integer too?

File: wsrep_config_service.h (hunk 1 above)
Comment (sciascid) [reply wsrep_config_service.h:89]: Some error handling is probably missing... will fix.

**Review by temeo — APPROVED** [review]

**Pull request merged**